rwengine/src/ai/CharacterController.cpp

```diff
@@ -30,9 +29,8 @@ bool CharacterController::updateActivity() {
     return false;
 }
 
-void CharacterController::setActivity(CharacterController::Activity *activity) {
-    if (_currentActivity) delete _currentActivity;
-    _currentActivity = activity;
+void CharacterController::setActivity(std::unique_ptr<Activity> activity) {
+    _currentActivity.swap(activity);
```

Comment:
why swap? Setting activity to _currentActivity seems redundant, could just be _currentActivity = std::move(activity);

Reply:
Btw Isn't it better to use static union or newer c++ style variant?